editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -0,0 +1,294 @@
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.wsdl2rest.ui.wizard;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.ISelectionService;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.PlatformUI;
+import org.fusesource.ide.camel.editor.utils.MavenUtils;
+import org.apache.maven.model.Dependency;
+import org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.fusesource.ide.wsdl2rest.ui.internal.UIMessages;
+import org.fusesource.ide.wsdl2rest.ui.internal.Wsdl2RestUIActivator;
+import org.fusesource.ide.wsdl2rest.ui.wizard.pages.Wsdl2RestWizardFirstPage;
+import org.fusesource.ide.wsdl2rest.ui.wizard.pages.Wsdl2RestWizardSecondPage;
+import org.jboss.fuse.wsdl2rest.impl.Wsdl2Rest;
+
+/**
+ * @author brianf
+ *
+ */
+public class Wsdl2RestWizard extends Wizard implements INewWizard {
+
+	/**
+	 * Collection of settings used by the wsdl2rest utility.
+	 */
+	final Wsdl2RestOptions options;
+	
+	private IProject project;
+
+	/**
+	 * Constructor
+	 */
+	public Wsdl2RestWizard() {
+		options = new Wsdl2RestOptions();
+	}
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.ui.IWorkbenchWizard#init(org.eclipse.ui.IWorkbench, org.eclipse.jface.viewers.IStructuredSelection)
+	 */
+	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		setWindowTitle(UIMessages.wsdl2RestWizardWindowTitle);
+		setNeedsProgressMonitor(true);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.jface.wizard.Wizard#performFinish()
+	 */
+	@Override
+	public boolean performFinish() {
+		try {
+			generate();
+		} catch (Exception e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
+			MessageDialog.openError(getShell(), "Error", e.getMessage());
```

Comment:
given that the iexception thrown is an encapsulated InvocationTargetException thrown by the Wsdl2Rest.generate method, the message will be always empty.

![image](https://user-images.githubusercontent.com/1105127/38995824-17895ef2-43ea-11e8-8012-6e1b212489de.png)

either improve the message in the encapsulated exception or just throw the original exception

Reply:
might be also a good idea to use the org.eclipse.jface.dialogs.ErrorDialog to have the stacktrace visible in UI if user wants to expand details.

Reply:
I provided a PR for this branch https://github.com/bfitzpat/fuseide/pull/23